### ISSUE TITLE
Implement a Notification that Strategies Have Been Automatically De-Selected 

### ISSUE THREAD

**Issue by entrotech** [hackforla/tdm-calculator#1285]
### Overview
When a user enters the project specifications on pages 1-3 of the Create Project Wizard and then selects some strategies on Page 4 (Strategies), then goes back to pages 1-3 and changes the project specifications, some of the strategies may no longer be applicable for the modified project. Issue #1284 implemented a change that automatically de-selects inapplicable strategies.  This issue is to provide a notification to users that strategies have been deselected.

### Action Items
- [x] Modify the calculation engine to return a true value from the run function if any strategies were deselected as describe above.
- [x] When the calculation engine returns a true value, display a modal dialog box informing the user that strategies were de-selected.  The wireframe can be found on Issue #1289 

### Resources/Instructions
Spawned by Project Management Issue #1265
Issue #1284 implements the automatic de-selection of inapplicable strategies

One sequence of steps to cause the de-selection to occur:
1. Enter a project with 35 Non-condominium dwelling units with 3 habitable rooms and 25000 sq ft of Retail space.
2. Specify 155 parking spaces on the Target Points Page.
3. On the Strategies Page, check the box for the Residential/Employment Package. 
4. Check that you now have a Level 1 project with 15 Target Points and 15 Earned Points.
5. Now go back to page 2 and remove the 35 residential units.  This makes the Package and its components no longer applicable, so the calculation will automatically de-select the Pricing/Unbundling strategy as well as the Package Bonus point to result in 6 Earned points.  Since the first part of this issue is completed, there should also be a toast message indicating that strategies have been de-selected.

**Comment by entrotech** [comment]
I implemented the first part to modify the engine.run function to return true if running the calculation resulted in deselection of one or more strategies.  As a placeholder, I modified TdmCalculationContainer to display a Toast message. 
To complete this issue, the Toast should be replaced with the modal dialog designed on Issue #1289.

Presumably, the wireframe proposed in the Dec 21, 2022 comment is the final design to be implemented for the Modal Dialog.
